### PR TITLE
[Finishes #163978010] Failing tests for user-login feature

### PR DESCRIPTION
**Pull request by maxwelbarno**
### What does this PR do?
Runs initial failing tests for the user-login feature

### Description of tasks to be completed
Failing user-login tests

### How should this be manually tested?
On your terminal, clone and execute the repository by running the following command
- Run ```git clone https://github.com/kipruto/politico-api.git```
- Run  ```cd politico-api```
- Run  ```source .env if dotenv is not installed```
- Run  ```pip install -r requirements.txt```
- Run  ```pytest --cov=app tests/v2```

### CODE REVIEW

File: tests/v2/test_auth.py [hunk 1]
```diff
@@ -13,6 +13,14 @@ def test_create_user(self):
...
         self.assertTrue(response_content['status'] == 201)
 
+    def test_user_login(self):
+        """Test that endpoint can login a registered user"""
+        super().create_user(admin_user)
+        response = super().login_user(admin_user_login)
+        self.assertEqual(response.status_code, 200)
+        response_content =  json.loads(response.data.decode())
+        self.assertTrue(response_content['status'] == 200)
+
```
Comment (petermwash) [inline tests/v2/test_auth.py:24]: Hi, Kindly add more tests for when a user tries to login with the wrong password, without an email, or without a password and when an unexisting user tries to log in

**Review by petermwash — CHANGES_REQUESTED** [review]

File: tests/v2/test_auth.py (hunk 1 above)
Comment (maxwelbarno) [reply tests/v2/test_auth.py:24]: Right away!

File: tests/v2/test_auth.py [hunk 2]
```diff
@@ -13,6 +13,54 @@ def test_create_user(self):
...
+    def test_blank_password_login(self):
+        """Test that endpoint can login a registered user"""
+        super().create_user(admin_user)
+        response = super().login_user(blank_password_login)
+        self.assertEqual(response.status_code, 400)
+        response_content =  json.loads(response.data.decode())
+        self.assertTrue(response_content['status'] == 400)
+
+    def test_blank_password_login(self):
```
Comment (petermwash) [inline tests/v2/test_auth.py:41]: this test is repeated as it is the same as the one above it.

File: tests/v2/test_auth.py [hunk 3]
```diff
@@ -13,6 +13,54 @@ def test_create_user(self):
...
+    def test_unexisting_login(self):
+        """Test that endpoint can login a registered user"""
+        super().create_user(admin_user)
+        response = super().login_user(blank_password_login)
+        self.assertEqual(response.status_code, 403)
+        response_content =  json.loads(response.data.decode())
+        self.assertTrue(response_content['status'] == 401)
+
+    def test_blank_body_login(self):
+        """Test that endpoint can login a registered user"""
```
Comment (petermwash) [inline tests/v2/test_auth.py]: The `doc strings` your tests are not describing what is happening in the methods, they are all talking about an `endpoint can log in a registered user` and yet the methods are testing different scenarios.

**Review by petermwash — CHANGES_REQUESTED** [review]

**Review by petermwash — APPROVED** [review]
LGTM.

**Comment by maxwelbarno** [comment]
Thanks.
Closed

**Pull request merged**